### ISSUE TITLE
NAACL-2019-Distant Supervision Relation Extraction with Intra-Bag and Inter-Bag Attentions

### ISSUE THREAD

**Issue by BrambleXu** [BrambleXu/knowledge-graph-learning#94]
**一句话总结：**

通过attention来调整bag representation的权重，对DS降噪。

**资源：**

- [code](<https://github.com/ZhixiuYe/Intra-Bag-and-Inter-Bag-Attentions>): pytorch 0.3 
- [pdf](<https://arxiv.org/abs/1904.00143>)

**关键字：**

- dataset: NYT
- DS noisy
- Previous studies mainly focus on sentence-level de-noising by designing neural networks with intra-bag attentions
- In this paper, both intrabag and inter-bag attentions are considered in order to deal with the noise at sentence-level and bag-level respectively.
- 

**笔记：**

First, relation-aware bag representations are calculated by
weighting sentence embeddings using intrabag attentions. Here, each possible relation is utilized as the query for attention calculation instead of only using the target relation in conventional methods. 

使用带有intrabag attentions的weighting sentence embeddings 来计算出relation-aware bag representations。

Furthermore, the representation of a group of bags in the training set which share the same relation label is calculated by weighting bag representations using a similarity-based inter-bag attention module. 

在一个group里面的bags，如果有相同的relation label，那么就通过inter-bag attention模型来调整不同bag representaiton的权重。

Finally, a bag group is utilized as a training sample when building our relation extractor. Experimental results on the New York Times dataset demonstrate the effectiveness of
our proposed intra-bag and inter-bag attention modules. 

最后，用a bag group当做训练样本来训练关系抽取器。

**模型图：**

![image-20190407100435937](https://ws2.sinaimg.cn/large/006tNc79gy1g1tsmeixmqj31i30u0tez.jpg)

**结果**：

![image-20190407100415781](https://ws1.sinaimg.cn/large/006tNc79gy1g1tsm1rkfaj30wk0ikjun.jpg)

![image-20190407100601408](https://ws2.sinaimg.cn/large/006tNc79gy1g1tsnvyny9j31ts0m445n.jpg)
